### PR TITLE
Federated signin bug fix

### PR DESCRIPTION
**Pull request by nidsharm**
*Issue #, if available:*
#388 #383 #346 

*Description of changes:*
bug fix for federated signin. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/aws-amplify/src/Auth/Auth.ts [hunk 1]
```diff
@@ -757,39 +757,52 @@ export default class AuthClass {
         return obj;
     }
 
-    private setCredentialsFromFederation(provider, token, user) {
+    private setCredentialsFromFederation(provider, token: string, user) {
         const domains = {
             'google': 'accounts.google.com',
             'facebook': 'graph.facebook.com',
             'amazon': 'www.amazon.com',
             'developer': 'cognito-identity.amazonaws.com'
```
Comment (mlabieniec) [inline packages/aws-amplify/src/Auth/Auth.ts]: add developer so it doesn't get rejected.

Comment (nidsharm) [reply packages/aws-amplify/src/Auth/Auth.ts]: Done.

File: packages/aws-amplify/src/Auth/Auth.ts [hunk 2]
```diff
@@ -757,39 +757,54 @@ export default class AuthClass {
...
-            user
-        );
-
-        if (AWS && AWS.config) { AWS.config.credentials = this.credentials; }
+        AWS.config.update({
+            region,
+            credentials : creds
+        });
+        
+        creds.getPromise().then(() => {
```
Comment (powerful23) [inline packages/aws-amplify/src/Auth/Auth.ts]: This async behavior may impact some methods in Auth category when using ```this.credentials``` immediately after calling this function.

**Comment by richardzcode** [comment]
I'm closing this one since too outdated.

**Pull request closed without merging**